sql/src/main/java/io/crate/planner/operators/TwoInputPlan.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.analyze.symbol.Symbol;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * A {@link LogicalPlan} with two other LogicalPlans as input.
+ */
+abstract class TwoInputPlan extends LogicalPlanBase {
```

Comment:
would change it to TwoInput**s**

Reply:
I'm not 100 % sure but singular might be correct here. Plural sounds strange to my ears. See also: https://scholar.google.de/scholar?hl=en&as_sdt=0%2C5&q=two+input&btnG=

Reply:
it's ok, not insisting ;-)